org.eclipse.xtext/src/org/eclipse/xtext/preferences/PreferenceValuesByLanguage.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.preferences;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.emf.common.notify.Adapter;
+import org.eclipse.emf.common.notify.Notifier;
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+
+public class PreferenceValuesByLanguage {
+	private final Map<String, IPreferenceValues> preferencesByLanguage = new HashMap<>();
+
+	public IPreferenceValues get(String languageId) {
+		return preferencesByLanguage.get(languageId);
+	}
+
+	public IPreferenceValues put(String languageId, IPreferenceValues values) {
+		return preferencesByLanguage.put(languageId, values);
+	}
+
+	public static PreferenceValuesByLanguage findInEmfObject(Notifier emfObject) {
+		for (Adapter adapter : emfObject.eAdapters())
+			if (adapter instanceof PreferenceValuesByLanguageAdapter)
+				return ((PreferenceValuesByLanguageAdapter) adapter).get();
+		return null;
+	}
+
+	public static PreferenceValuesByLanguage removeFromEmfObject(Notifier emfObject) {
+		List<Adapter> adapters = emfObject.eAdapters();
+		for (int i = 0, max = adapters.size(); i < max; i++) {
```

Comment:
This max isn't needed, we just need to use an iterator.

Reply:
Will remove the max.